whylog/constraints/verifier.py

```diff
@@ -0,0 +1,177 @@
+import itertools
+
+from whylog.config import Clue
+from whylog.constraints import IdenticalIntervals, TimeConstraint
+from whylog.constraints.exceptions import UnsupportedConstraintTypeError
+from whylog.front import FrontInput
+
+
+class ConstraintRegistration(object):
+    CONSTRAINTS = {
+        'identical': IdenticalIntervals({}),
+        'time': TimeConstraint({})
+        # register your constraint here
+    }
+
+    @classmethod
+    def get_constraint(cls, constraint_data):
+        if constraint_data['name'] in cls.CONSTRAINTS:
+            return cls.CONSTRAINTS[constraint_data['name']]
+        raise UnsupportedConstraintTypeError(constraint_data)
+
+
+class ConstraintManager(object):
+    """
+    there should be one such object per rule being verified
+    """
+
+    def __init__(self):
+        self._actual_constraints = {}
+
+    def __getitem__(self, constraint_data):
+        if not constraint_data['name'] in self._actual_constraints:
```

Comment:
lookup is performed twice in this function. Use `constraint = self._actual_constraints.get(constraint_data['name'])` and check `if constraint is None`